Social | Add post list page skeleton UI

manzoorwanijk opened the pull request:
Now that we are starting to work on the post list UI for Social to add scheduled resharing, we need to decouple certain components and the store logic. In order to test that decoupling, we need a functional UI. That is what this PR does.

## Proposed changes:
<!--- Explain what functional changes your PR includes -->
* Create a basic functional UI for Social on the post list page, hidden behind a feature flag.

## Implementation

The modal is created only once in the admin footer and then re-used for each post when "Share" button is clicked.

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [ ] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion
<!-- If you're an Automattician, include a shortlink to the p2 discussion with Jetpack Product here. -->
<!-- Make sure any changes to existing products have been discussed and agreed upon -->

## Does this pull request change what data or activity we track or use?
<!--- If so, please add the "[Status] Needs Privacy Updates" label and explain what changes there are. -->
<!--- Check existing Jetpack support documents for a preview of the information we need. -->

## Testing instructions:
<!-- If you were reviewing this PR, how would you like the instructions to be presented? -->
<!-- Please include detailed testing steps, explaining how to test your change. -->
<!-- Bear in mind that context you working on is not obvious for everyone.  -->
<!-- Adding "simple" configuration steps will help reviewers to get to your PR as quickly as possible. -->
<!-- "Before / After" screenshots can also be very helpful when the change is visual. -->

* Run `jetpack build packages/publicize`
* Go to post list page
* Confirm that there is no "Share" button for any post, whether published or draft
* Now add this constant to `wp-config.php`
```php
define( 'JETPACK_SOCIAL_HAS_POST_LIST_UI', true );
```
* Go to post list page again
* Confirm that you see the "Share" button for published posts but not for drafts
* Click on "Share"
* Confirm that it opens the basic modal with correct post ID
* Close the modal
* Click "Share" for a different post
* Confirm that there are no warnings in the console about non-existent root

## Screenshots
- Share button for published posts but not for drafts
    <img width="433" alt="Screenshot 2025-03-04 at 11 41 52 AM" src="https://github.com/user-attachments/assets/e8fb40fe-a384-4a9a-b75d-cb7674c33631" />
- The share post modal
    <img width="395" alt="Screenshot 2025-03-04 at 11 42 39 AM" src="https://github.com/user-attachments/assets/6975924a-83c4-46df-bfd3-b39e078303d0" />

manzoorwanijk committed:
Create a basic App component

manzoorwanijk committed:
Create an entry point to render the basic component

manzoorwanijk committed:
Create webpack entrypoint for the bundle

manzoorwanijk committed:
Hook it all up from the backend

manzoorwanijk committed:
Add changelog

manzoorwanijk committed:
Fix unit tests

manzoorwanijk committed:
Use current screen instead of globals

manzoorwanijk committed:
Use static methods for re-usability outsite the class

pablinos commented on projects/packages/publicize/src/class-post-list-page.php:
```diff
@@ -0,0 +1,138 @@
+/**
+ * The class to handle the Post List Page.
+ */
+class Post_List_Page {
```
There is [similar code](https://github.com/Automattic/jetpack/blob/e932d72a34cffd1ad5ec914ea68bddfe0d6d73fe/projects/packages/post-list/src/class-post-list.php#L133-L153) to this which renders the current 'Share' link and opens the editor. I'm not sure if that post-list package still makes sense, but it is enabled on wpcom, so we could ship this feature via it still.

pablinos commented on projects/packages/publicize/src/class-post-list-page.php:
```diff
@@ -0,0 +1,138 @@
+		}
+
+		/** This filter is documented in projects/packages/publicize/src/class-publicize-base.php */
+		$capability = apply_filters( 'jetpack_publicize_capability', 'publish_posts' );
```
I don't think we're checking this with the current 'Share' link on the post list screen. We should make sure that we do! It's probably not a particular problem currently as it opens the editor with the Jetpack sidebar open, so the capability will be checked at that point.

pablinos commented on projects/packages/publicize/src/class-post-list-page.php:
```diff
@@ -0,0 +1,138 @@
+			return;
+		}
+
+		Assets::register_script(
+			'jetpack-social-post-list-page',
+			'../build/post-list-page.js',
+			__FILE__,
+			array(
+				'in_footer'  => true,
+				'textdomain' => 'jetpack-publicize-pkg',
+				'enqueue'    => true,
+			)
+		);
```
I don't know if we we would want to do this, or add our script to the current post-list package one (it will probably just rendering the modal).

pablinos commented on projects/packages/publicize/src/class-post-list-page.php:
```diff
@@ -0,0 +1,138 @@
+		if ( ! Publicize_Script_Data::has_feature_flag( 'post-list-ui' ) ) {
+			return;
+		}
+		add_action( 'admin_footer', array( $this, 'render_share_post_root' ) );
+		add_filter( 'post_row_actions', array( $this, 'append_share_post_action' ), 10, 2 );
+		add_action( 'admin_enqueue_scripts', array( $this, 'enqueue_admin_scripts' ) );
```
What about moving these to an admin_init hook, but not adding them if should_render_on_page returns false? That stops us from having to call should_render_on_page more than once.

But we might also me amalgamating this with the post-list package version, so we can decide on the code structure, once we decide on that.

manzoorwanijk replied:
If we want to use `get_current_screen`, we can't do that before `current_screen` because `admin_init` happens before that. So, using `current_screen` instead of `admin_init` is a better option.

pablinos reviewed (commented):
I think this might not be needed right now, but having the skeleton UI and entrypoint setup could be useful.

I'm mainly wondering about how we use (or not) the post-list package, which renders the current Share link.

We also might stick to taking people into the editor for now. In which case we'll want to keep the Share link we currently have (perhaps changing the query parameter).

manzoorwanijk commented:
I have converted this to a draft to focus on the editor UI for this one for now.

manzoorwanijk committed:
Merge branch 'trunk' into add/social/post-list-page-skeleton-ui

manzoorwanijk committed:
Update pnpm-lock.yaml

manzoorwanijk committed:
Use current_screen hook for early check on the page

manzoorwanijk committed:
Retain the existing data from post-list package

Once we this is ready, we can get rid of the share list from post-list package.

manzoorwanijk committed:
Add republicize feature check

manzoorwanijk committed:
Phan is not happy with esc_attr on $post->ID

manzoorwanijk committed:
Merge branch 'trunk' into add/social/post-list-page-skeleton-ui
